src/Common/src/System/Security/Cryptography/DerEncoder.cs

```diff
@@ -445,6 +445,74 @@ internal static byte[][] SegmentedEncodeOid(Oid oid)
         }
 
         /// <summary>
+        /// Encode a string as a UTF8String value.
+        /// </summary>
+        /// <param name="str">The string whose to be encoded.</param>
+        /// <returns>The encoded segments { tag, length, value }</returns>
+        internal static byte[][] SegmentedEncodeUtf8String(string str)
+        {
+            Debug.Assert(str != null);
+
+            byte[] encodedBytes = System.Text.Encoding.UTF8.GetBytes(str);
+
+            return new byte[][]
+            {
+                new byte[] { (byte)DerSequenceReader.DerTag.UTF8String },
+                EncodeLength(encodedBytes.Length),
+                encodedBytes,
+            };
+        }
+
+        /// <summary>
+        /// Encode a substring as a UTF8String value.
+        /// </summary>
+        /// <param name="str">The string whose substring is to be encoded.</param>
+        /// <param name="offset">The character offset into <paramref name="str"/> at which to start.</param>
+        /// <param name="count">The total number of characters from <paramref name="str"/> to read.</param>
+        /// <returns>The encoded segments { tag, length, value }</returns>
+        internal static byte[][] SegmentedEncodeUtf8String(string str, int offset, int count)
+        {
+            Debug.Assert(str != null);
+            Debug.Assert(offset >= 0);
+            Debug.Assert(offset <= str.Length);
+            Debug.Assert(count >= 0);
+            Debug.Assert(count <= str.Length - offset);
+
+            int encodedSize;
+
+            // ITU-T X.690 says that ISO/IEC 10646 Annex D should be used; but no announcers
+            // or escape sequences, and each character shall be encoded in the smallest number of
+            // bytes possible.
+            //
+            // Thankfully, that's just Encoding.UTF8.
+
+            unsafe
```

Comment:
I'm not a fan of unsafe code when there's a safe equivalent and there's not an important reason for it. Can you switch to the char[] overload and just use safe code here?